### PR TITLE
feat(api): In org details, show if SSO is required

### PR DESCRIPTION
**Pull request by RyanSkonnord**
Supports https://github.com/getsentry/sentry/pull/73521

### CODE REVIEW

**Review by cathteng — APPROVED** [review]

**Review by iamrajjoshi — APPROVED** [review]
I am not sure if we need to update our API docs for this change.

File: src/sentry/api/serializers/models/organization.py [hunk 1]
```diff
@@ -598,6 +599,7 @@ def serialize(  # type: ignore[explicit-override, override]
         if access.role is not None:
             context["role"] = access.role  # Deprecated
             context["orgRole"] = access.role
+        context["requiresSso"] = access.requires_sso
```
Comment (evanpurkhiser) [inline src/sentry/api/serializers/models/organization.py:602]: Is there another way we can make this check without having to update the org serializer? I'm sure we must know somewhere on the frontend if the org requires SSO no?

Comment (evanpurkhiser) [reply src/sentry/api/serializers/models/organization.py:602]: Chatted with @RyanSkonnord about this. We really don't expose this anywhere to the UI for the most part. I think adding this here is fine (despite it being just another field on an organization).

**Review by evanpurkhiser — APPROVED** [review]

File: src/sentry/api/serializers/models/organization.py (hunk 1 above)
Comment (RyanSkonnord) [reply src/sentry/api/serializers/models/organization.py:602]: Some notes on my investigation, for the record:

The source of truth for `access.requires_sso` is `query_sso_state` at https://github.com/getsentry/sentry/blob/master/src/sentry/auth/services/access/service.py#L60, which inspects the associated `AuthProvider` object. The flag is generally available through an `Access` object and isn’t associated directly with the org model. Given that the org serializer is already adding few other `Access` attributes to the API output (right above here), there seems to be precedent. However, I did have the same initial intuition as @evanpurkhiser — it's surprising that it wasn’t already visible somehow.

**Comment by RyanSkonnord** [comment]
> I am not sure if we need to update our API docs for this change.

According to @isabellaenriquez an update would be overwritten by https://github.com/getsentry/sentry/pull/73293 anyway, so we can add it afterward.

**Pull request merged**